app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/ConfigurableVariant.php

```diff
@@ -92,19 +79,15 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
         $this->variantCollection->addEavAttributes($matchedFields);
         $this->optionCollection->addProductId((int)$value[$linkField]);
 
-        $result = function () use ($value, $linkField) {
-            $children = $this->variantCollection->getChildProductsByParentId((int)$value[$linkField]);
-            $options = $this->optionCollection->getAttributesByProductId((int)$value[$linkField]);
-            $variants = [];
-            /** @var Product $child */
-            foreach ($children as $key => $child) {
-                $variants[$key] = ['sku' => $child['sku'], 'product' => $child, 'options' => $options];
-            }
-
-            return $variants;
-        };
+        $children = $this->variantCollection->getChildProductsByParentId((int)$value[$linkField]);
```

Comment:
This was not supposed to be replaced because this was actual performance optimization via Deferred mechanism.

Reply:
Fixed